src/Payment.ts

```diff
@@ -262,11 +275,32 @@ export class Payment {
    * Signs a raw transaction object as returned by `prepare`
    * and sends the signed transaction.
    * @param rawTx Raw transaction object.
+   * @deprecated use confirmPayment instead.
    */
   public async confirm(rawTx: RawTxObject): Promise<any> {
     return this.transaction.confirm(rawTx)
   }
 
+  /**
+   * Signs the rawTx provided as returned by `prepare`
+   * and sends the signed transaction as well as the message with messageId
+   * @param rawTx Raw transaction object.
+   * @param messageId The messageId returned when preparing a payment with message
+   * @param message The message to be sent
+   */
+  public async confirmPayment(
+    { rawTx, messageId }: { rawTx: RawTxObject; messageId?: string },
```

Comment:
Since I did not give a name to this object, there is no other way to document it.
The function is just meant to receive an object with `rawTx` and `messageId` and a `message`. The reason I do that is to make it as easy to use as possible for the mobile app, as you can see in the test you can do:
```
const preparedPayment = await prepare(...)
await confirmPayment(preparedPayment, 'message')
```

Reply:
Okay it is this way. I would argue that is is fine to use the `PaymentTxObject` as parameter, since this is the user flow with the clientlib. But the docs still don't tell the user that it is like this. But if you insist to keep this signature nominalistic just close this conversation.